pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -472,9 +472,16 @@ func (e *EndpointController) syncService(key string) error {
 
 	createEndpoints := len(currentEndpoints.ResourceVersion) == 0
 
+	// Compare the sorted subsets and labels
+	// Remove the HeadlessService label from the endpoints if it exists, as this won't be set on the service itself
```

Comment:
nit: can you please split the lines to make the shorter?

Reply:
done